### PR TITLE
fix display of remote users in incoming share notifications

### PR DESCRIPTION
**Pull request by blizzz**
1. Have two Nextcloud instances with LDAP and User IDs from UUIDs
2. Make those instances trusted and ensure that addressbooks are exchanged
3. Do a remote share from one instance to the other
4. As recipient, have a look at your notifications

Before

Screenshot_20200730_211335

After:

Screenshot_20200730_211355

**Hint for reviewers**: have a look at the first commit, the second contains code style changes only.

### CODE REVIEW

File: apps/federatedfilesharing/lib/Notifier.php [hunk 1]
```diff
@@ -213,17 +210,24 @@ protected function getDisplayName(ICloudId $cloudId) {
...
 		}
 
 		try {
-			$this->getDisplayNameFromContact($user . '@https://' . $server);
+			return $this->getDisplayNameFromContact($user . '@http://' . $server);
+		} catch (\OutOfBoundsException $e) {
+		}
+
+		try {
+			return $this->getDisplayNameFromContact($user . '@https://' . $server);
```
Comment (nickvergessen) [inline apps/federatedfilesharing/lib/Notifier.php:230]: https first? :)

File: apps/dav/lib/CardDAV/CardDavBackend.php [hunk 2]
```diff
@@ -974,8 +974,9 @@ public function search($addressBookId, $pattern, $searchProperties, $options = [
 					continue;
 				}
 
-				if ($property === 'CLOUD' && preg_match('/[^a-zA-Z0-9 _.@\-\']/', $pattern) === 1) {
-					// There can be no chars in cloud ids which are not valid for user ids
+				if ($property === 'CLOUD' && preg_match('/[^a-zA-Z0-9 :_.@\/\-\']/', $pattern) === 1) {
```
Comment (nickvergessen) [inline apps/dav/lib/CardDAV/CardDavBackend.php:977]: D'oh!

File: apps/dav/lib/CardDAV/CardDavBackend.php [hunk 3]
```diff
@@ -159,13 +159,13 @@ public function getAddressBooksForUser($principalUri) {
...
 			$addressBooks[$row['id']] = [
-				'id'  => $row['id'],
+				'id' => $row['id'],
 				'uri' => $row['uri'],
 				'principaluri' => $this->convertPrincipal($row['principaluri'], false),
 				'{DAV:}displayname' => $row['displayname'],
 				'{' . Plugin::NS_CARDDAV . '}addressbook-description' => $row['description'],
 				'{http://calendarserver.org/ns/}getctag' => $row['synctoken'],
-				'{http://sabredav.org/ns}sync-token' => $row['synctoken']?$row['synctoken']:'0',
+				'{http://sabredav.org/ns}sync-token' => $row['synctoken'] ? $row['synctoken'] : '0',
```
Comment (nickvergessen) [inline apps/dav/lib/CardDAV/CardDavBackend.php:168]:
```suggestion
				'{http://sabredav.org/ns}sync-token' => $row['synctoken'] ?: '0',
```

File: apps/dav/lib/CardDAV/CardDavBackend.php [hunk 4]
```diff
@@ -215,13 +215,13 @@ public function getAddressBooksForUser($principalUri) {
...
 			$addressBooks[$row['id']] = [
-				'id'  => $row['id'],
+				'id' => $row['id'],
 				'uri' => $uri,
 				'principaluri' => $principalUriOriginal,
 				'{DAV:}displayname' => $displayName,
 				'{' . Plugin::NS_CARDDAV . '}addressbook-description' => $row['description'],
 				'{http://calendarserver.org/ns/}getctag' => $row['synctoken'],
-				'{http://sabredav.org/ns}sync-token' => $row['synctoken']?$row['synctoken']:'0',
+				'{http://sabredav.org/ns}sync-token' => $row['synctoken'] ? $row['synctoken'] : '0',
```
Comment (nickvergessen) [inline apps/dav/lib/CardDAV/CardDavBackend.php:224]:
```suggestion
				'{http://sabredav.org/ns}sync-token' => $row['synctoken'] ?: '0',
```

File: apps/dav/lib/CardDAV/CardDavBackend.php [hunk 5]
```diff
@@ -237,21 +237,21 @@ public function getUsersOwnAddressBooks($principalUri) {
...
 			$addressBooks[$row['id']] = [
-				'id'  => $row['id'],
+				'id' => $row['id'],
 				'uri' => $row['uri'],
 				'principaluri' => $this->convertPrincipal($row['principaluri'], false),
 				'{DAV:}displayname' => $row['displayname'],
 				'{' . Plugin::NS_CARDDAV . '}addressbook-description' => $row['description'],
 				'{http://calendarserver.org/ns/}getctag' => $row['synctoken'],
-				'{http://sabredav.org/ns}sync-token' => $row['synctoken']?$row['synctoken']:'0',
+				'{http://sabredav.org/ns}sync-token' => $row['synctoken'] ? $row['synctoken'] : '0',
```
Comment (nickvergessen) [inline apps/dav/lib/CardDAV/CardDavBackend.php:254]:
```suggestion
				'{http://sabredav.org/ns}sync-token' => $row['synctoken'] ?: '0',
```

File: apps/dav/lib/CardDAV/CardDavBackend.php [hunk 6]
```diff
@@ -292,13 +292,13 @@ public function getAddressBookById($addressBookId) {
...
 		$addressBook = [
-			'id'  => $row['id'],
+			'id' => $row['id'],
 			'uri' => $row['uri'],
 			'principaluri' => $row['principaluri'],
 			'{DAV:}displayname' => $row['displayname'],
 			'{' . Plugin::NS_CARDDAV . '}addressbook-description' => $row['description'],
 			'{http://calendarserver.org/ns/}getctag' => $row['synctoken'],
-			'{http://sabredav.org/ns}sync-token' => $row['synctoken']?$row['synctoken']:'0',
+			'{http://sabredav.org/ns}sync-token' => $row['synctoken'] ? $row['synctoken'] : '0',
```
Comment (nickvergessen) [inline apps/dav/lib/CardDAV/CardDavBackend.php:301]:
```suggestion
			'{http://sabredav.org/ns}sync-token' => $row['synctoken'] ?: '0',
```

File: apps/dav/lib/CardDAV/CardDavBackend.php [hunk 7]
```diff
@@ -326,13 +326,13 @@ public function getAddressBooksByUri($principal, $addressBookUri) {
...
```
Comment (nickvergessen) [inline apps/dav/lib/CardDAV/CardDavBackend.php:335]:
```suggestion
			'{http://sabredav.org/ns}sync-token' => $row['synctoken'] ?: '0',
```

File: apps/federatedfilesharing/lib/Notifier.php [hunk 8]
```diff
@@ -107,9 +107,7 @@ public function prepare(INotification $notification, string $languageCode): INot
 					);
 
 					$initiator = $params[0];
-					$initiatorDisplay = isset($params[3]) ? $params[3] : null;
```
Comment (nickvergessen) [inline apps/federatedfilesharing/lib/Notifier.php:110]: then we can revert the change where this was added?

**Review by nickvergessen — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

File: apps/federatedfilesharing/lib/Notifier.php (hunk 1 above)
Comment (blizzz) [reply apps/federatedfilesharing/lib/Notifier.php:230]: didn't touch the order. can switch. actually i do wonder whether there is a valid use case for those two fall backs?

File: apps/federatedfilesharing/lib/Notifier.php (hunk 8 above)
Comment (blizzz) [reply apps/federatedfilesharing/lib/Notifier.php:110]: some details (formating, error information) are still in place.

**Comment by juliushaertl** [comment]
Any reason not to backport this to stable18 ?

**Comment by blizzz** [comment]
> Any reason not to backport this to stable18 ?

I don't remember if it was affected or not. If it is, and the fix applies, then nothing speaks against a backport.

**Comment by ChristophWurst** [comment]
/backport b79e34c57ba74246a03a66045a6b792e35da1d32 ac3d8d1e43766c1912f95cf0638548cdba1a6ab2 79027c8eff1a2b4c88924e6d34343581b7e10605 to stable18
